plugins/inputs/pubsub/pubsub.go

```diff
@@ -0,0 +1,291 @@
+package pubsub
```

Comment:
We should rename this plugin to qualify it further since we also want would support other pub/sub systems and the name `pubsub` is too generic.  A couple name ideas I have are `cloud_pubsub`, `google_pubsub`, `google_cloud_pubsub`.

We should also consider how a push delivery endpoint should be named (#4700).  Since these two methods don't have much shared configuration and the API is completely different, I think they should remain separate plugins.  With this in mind maybe we should have `cloud_pubsub_pull` and `cloud_pubsub_push`?  If you think it is better to emphasize one of these more then maybe it should lopsided: `cloud_pubsub` and `cloud_pubsub_push`.

Reply:
I think google_pubsub and google_pubsub_push work - @gunnar @josephlewis42 opinions?

Reply:
I think Google's branding guidelines would recommend `cloud_pubsub` or `google_cloud_pubsub`.

I like the idea of the lopsided pull vs push, especially with pull being the more common case. That should get them sorted nicely lexicographically for documentation too.

The hardest part of any project is naming :grin: